src/main/java/org/candlepin/subscriptions/tally/export/InstancesCsvDataMapperService.java

```diff
@@ -88,4 +111,44 @@ public Class<TallyInstanceView> getDataClass() {
   public Class<?> getExportItemClass() {
     return InstancesExportCsvItem.class;
   }
+
+  private boolean needsToExportGuests(TallyInstanceView item) {
+    return item.getNumOfGuests() > 0
+        && RHEL_FOR_X86.equals(item.getKey().getProductId())
+        && HardwareMeasurementType.HYPERVISOR.equals(item.getKey().getMeasurementType());
+  }
+
+  private List<InstancesExportCsvItem> mapGuests(TallyInstanceView item) {
+    var guestsInInstance =
+        getGuestHostsByHypervisorInstanceId(item, PageRequest.ofSize(MAX_GUESTS_PER_QUERY));
+    Set<InstancesExportCsvItem> guests =
+        new HashSet<>(toInstancesExportCsvItem(guestsInInstance.getContent()));
+    while (guestsInInstance.hasNext()) {
+      guestsInInstance = getGuestHostsByHypervisorInstanceId(item, guestsInInstance.nextPageable());
+      guests.addAll(toInstancesExportCsvItem(guestsInInstance.getContent()));
+    }
+
+    return new ArrayList<>(guests);
+  }
+
+  private List<InstancesExportCsvItem> toInstancesExportCsvItem(List<Host> guests) {
+    return guests.stream()
+        .map(
+            guest ->
+                new InstancesExportCsvItem()
+                    .withDisplayName(guest.getDisplayName())
+                    .withInventoryId(guest.getInventoryId())
+                    .withHardwareType(
```

Comment:
Actually it'd be ideal to include this for all hosts.

Reply:
The problem is that we would need to add it to the view which would require a lot of more changes. 
Also, this hardware type is not included in the Instances API, so I'd prefer not to include it unless it's requested as part of another ticket.